From 58f0df44f5da3f83a5c0dc1b554ddf2d7099046e Mon Sep 17 00:00:00 2001
From: Martin Pitt <martin.pitt@ubuntu.com>
Date: Thu, 8 Sep 2011 09:14:02 +0200
Subject: [PATCH] Consistently treat gdkenumtypes.[hc] as source files

Always generate gdk/gdkenumtypes.c in the source tree, and fix path in the
introspection sources. With this, gdkenumtypes.c is correctly included with
separate build trees, too. This fixes missing GTypes in the .gir when using a
separate build tree.

Bug: https://bugzilla.gnome.org/show_bug.cgi?id=647729
Bug-Ubuntu: https://launchpad.net/bugs/769256
---
 gdk/Makefile.am |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/gdk/Makefile.am b/gdk/Makefile.am
index c7f24af..e7972d4 100644
--- a/gdk/Makefile.am
+++ b/gdk/Makefile.am
@@ -185,8 +185,8 @@ if HAVE_INTROSPECTION
 introspection_files = 		\
 	$(filter-out gdkkeysyms-compat.h, $(gdk_public_h_sources))	\
 	$(gdk_c_sources)	\
-        gdkenumtypes.c		\
-        gdkenumtypes.h
+        $(srcdir)/gdkenumtypes.c	\
+        $(srcdir)/gdkenumtypes.h
 
 Gdk-3.0.gir: libgdk-3.la Makefile
 Gdk_3_0_gir_SCANNERFLAGS = 	\
@@ -320,13 +320,13 @@ gdkenumtypes.h: stamp-gdkenumtypes.h
 stamp-gdkenumtypes.h: @REBUILD@ $(gdk_public_h_sources) gdkenumtypes.h.template
 	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gdkenumtypes.h.template \
 		$(gdk_public_h_sources) ) >> xgen-geth \
-	&& (cmp -s xgen-geth gdkenumtypes.h || cp xgen-geth gdkenumtypes.h ) \
+	&& (cmp -s xgen-geth gdkenumtypes.h || cp xgen-geth $(srcdir)/gdkenumtypes.h ) \
 	&& rm -f xgen-geth \
 	&& echo timestamp > $(@F)
 gdkenumtypes.c: @REBUILD@ $(gdk_public_h_sources) gdkenumtypes.c.template
 	$(AM_V_GEN) ( cd $(srcdir) && $(GLIB_MKENUMS) --template gdkenumtypes.c.template \
 		$(gdk_public_h_sources) ) > xgen-getc \
-	&& cp xgen-getc gdkenumtypes.c  \
+	&& cp xgen-getc $(srcdir)/gdkenumtypes.c  \
 	&& rm -f xgen-getc
 
 #
-- 
1.7.5.4

